wsimod/nodes/nodes.py

```diff
@@ -88,6 +88,16 @@ def apply_overrides(self, overrides: Dict[str, Any] = {}) -> None:
         Args:
             overrides (dict, optional): Dictionary of overrides. Defaults to {}.
         """
+        # overrides data_input_dict
+        from wsimod.orchestration.model import read_csv
+        content = overrides.pop('data_input_dict', self.data_input_dict)
+        if isinstance(content, str):
+            self.data_input_dict = read_csv(content)
+        elif isinstance(content, dict):
+            self.data_input_dict = content
+        else:
```

Comment:
Maybe I misunderstand but don't we need something in case they are not overriding `data_input_dict` or that node doesn't use `data_input_dict` ?

```suggestion
        elif not content:
                pass
        else:
```